controllers/machine_controller.go

```diff
@@ -189,3 +227,91 @@ func makeMachineInfo(rawInfo *naglfarv1.MachineInfo) (*naglfarv1.MachineInfo, er
 
 	return info, nil
 }
+
+func (r *MachineReconciler) tryPullImage(dockerClient docker.APIClient) error {
+	_, _, err := dockerClient.ImageInspectWithRaw(r.Ctx, machineWorkerImage)
```

Comment:
This code snippet occurs many times(like TestResourceReconciler.pullImageByPolicy), would you extract it to a method?

Reply:
I would do it if we remove events from `TestResourceReconciler.pullImageByPolicy`, otherwise there will be to many arguments.

Reply:
Fine, I think events can move to the caller method.

Reply:
Fixed in [094a073](https://github.com/PingCAP-QE/Naglfar/pull/58/commits/094a073a5c98cb34c5aa495ed6d059fbf2c006b5).